lib/csharp/test/TestIOStreamServer/TestIOStreamClient.cs

```diff
@@ -0,0 +1,425 @@
+
+        [TestMethod]
+        public void testStruct()
+        {
+            Xtruct o = new Xtruct();
+            o.String_thing = "Zero";
+            o.Byte_thing = (sbyte)1;
+            o.I32_thing = -3;
+            o.I64_thing = -5;
+            Xtruct i = client.testStruct(o);
+            Assert.AreEqual(o.String_thing, i.String_thing);
+            Assert.AreEqual(o.Byte_thing, i.Byte_thing);
+            Assert.AreEqual(o.I32_thing, i.I32_thing);
+            Assert.AreEqual(o.I64_thing, i.I64_thing);
+        }
+
+        [TestMethod]
+        public void testNest()
+        {
+            Xtruct o = new Xtruct();
+            o.String_thing = "Zero";
+            o.Byte_thing = (sbyte)1;
+            o.I32_thing = -3;
+            o.I64_thing = -5;
+
+            Xtruct2 o2 = new Xtruct2();
+            o2.Byte_thing = (sbyte)1;
+            o2.Struct_thing = o;
+            o2.I32_thing = 5;
+            Xtruct2 i2 = client.testNest(o2);
+            Assert.AreEqual(o2.Byte_thing, i2.Byte_thing);
+            Assert.AreEqual(o2.Struct_thing.String_thing, i2.Struct_thing.String_thing);
+            Assert.AreEqual(o2.Struct_thing.Byte_thing, i2.Struct_thing.Byte_thing);
+            Assert.AreEqual(o2.Struct_thing.I32_thing, i2.Struct_thing.I32_thing);
+            Assert.AreEqual(o2.Struct_thing.I64_thing, i2.Struct_thing.I64_thing);
+            Assert.AreEqual(o2.I32_thing, i2.I32_thing);
+        }
+
+        [TestMethod]
+        public void testMap()
+        {
+            Dictionary<int, int> mapout = new Dictionary<int, int>();
+            for (int j = 0; j < 5; j++)
+            {
+                mapout[j] = j - 10;
+            }
+
+            Dictionary<int, int> mapin = client.testMap(mapout);
+            Assert.AreEqual(mapout.Count, mapin.Count);
+            foreach (KeyValuePair<int, int> pair in mapout)
+            {
+                Assert.AreEqual(pair.Value, mapin[pair.Key]);
+            }
+        }
+
+        [TestMethod]
+        public void testList()
+        {
+            List<int> listout = new List<int>();
+            for (int j = -2; j < 3; j++)
+            {
+                listout.Add(j);
+            }
+
+            List<int> listin = client.testList(listout);
+
+            Assert.AreEqual(listout.Count, listin.Count);
+            foreach (int i in listout)
+            {
+                Assert.IsTrue(listin.Contains(i));
+            }
+        }
+
+        [TestMethod]
+        public void testSet()
+        {
+            THashSet<int> setout = new THashSet<int>();
+            for (int j = -2; j < 3; j++)
+            {
+                setout.Add(j);
+            }
+
+            THashSet<int> setin = client.testSet(setout);
+
+            Assert.AreEqual(setout.Count, setin.Count);
+            foreach (int i in setout)
+            {
+                Assert.IsTrue(setin.Contains(i));
+            }
+        }
+
+        [TestMethod]
+        public void testEnum()
+        {
```

Comment:
I'd rather like the test being included into the existing standard ThriftTest test client, instead of writing a duplicated one. Having two of them does not make much sense, do you agree?